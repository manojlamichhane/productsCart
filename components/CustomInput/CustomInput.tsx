import React from "react";

const CustomInput = (props: {
  label: string;
  name: string;
  placeholder: string;
  value: string;
  handleChange: any;
}) => {
  return (
    <div>
      <label className="block mb-2 text-sm font-medium text-black ">
        {props.label}
      </label>
      <input
        type="text"
        name={props.name}
        id={props.name}
        data-testid="input"
        value={props.value}
        onChange={props.handleChange}
        className=" border border-gray-300 text-black sm:text-sm rounded-lg  block w-full p-2.5 "
        placeholder={props.placeholder}
        required={true}
      />
    </div>
  );
};

export default CustomInput;
