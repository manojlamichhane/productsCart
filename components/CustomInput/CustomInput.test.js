import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import CustomInput from "./CustomInput";

describe("Input test", () => {
  test("Render check", () => {
    const { asFragment } = render(<CustomInput />);
    expect(screen.getByTestId("input")).toBeInTheDocument();
    expect(screen.getByRole("textbox")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
  test("Should render with given placeholder", () => {
    render(<CustomInput placeholder="user" />);
    expect(screen.getByPlaceholderText("user")).toBeInTheDocument();
  });
  test("Should render with given value", () => {
    const handleChange = jest.fn();
    render(<CustomInput value="User" handleChange={handleChange} />);
    expect(screen.getByDisplayValue("User")).toBeInTheDocument();
  });
});
