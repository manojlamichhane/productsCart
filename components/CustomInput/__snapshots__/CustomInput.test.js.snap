// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test Render check 1`] = `
<DocumentFragment>
  <div>
    <label
      class="block mb-2 text-sm font-medium text-black "
    />
    <input
      class=" border border-gray-300 text-black sm:text-sm rounded-lg  block w-full p-2.5 "
      data-testid="input"
      required=""
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
